File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i = 0
        max_length = 0
        for j, char in enumerate(s):
            while char in s[i:j]:
                i = i + 1
            max_length = max(max_length, j - i + 1)
        return max_length

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring =     function (s){
    // add whatever parameters you deem necessary - good luck!
    let left = 0;
    let longest_substr_length = 0;
    while(left < s.length){
        const hash_map = {};
        let ptr = left;
        while(!hash_map[s[ptr]]){
            hash_map[s[ptr]] = 1;
            ptr += 1;
            if(ptr == s.length){
                break;
            }
        }
        longest_substr_length = Math.max(longest_substr_length, ptr - left);
        left += 1;
    }
    return longest_substr_length;
}